test/e2e/reconciler/workspace/apibinding_initializer_test.go

```diff
@@ -160,3 +162,85 @@ func TestWorkspaceTypesAPIBindingInitialization(t *testing.T) {
 	// This will create and wait for ready, which only happens if the APIBinding initialization is working correctly
 	_, _ = framework.NewWorkspaceFixture(t, server, universalPath, framework.WithType(universalPath, "test"), framework.WithName("init"))
 }
+
+func TestWorkspaceTypesAPIBindingWithCustomPermissions(t *testing.T) {
+	t.Parallel()
+	framework.Suite(t, "control-plane")
+
+	server := framework.SharedKcpServer(t)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+
+	orgPath, _ := framework.NewOrganizationFixture(t, server)
+
+	cowboysProviderPath, _ := framework.NewWorkspaceFixture(t, server, orgPath, framework.WithName("cowboys-provider"))
+
+	cfg := server.BaseConfig(t)
+	kcpClusterClient, err := kcpclientset.NewForConfig(cfg)
+	require.NoError(t, err, "error creating kcp cluster client")
+
+	dynamicClusterClient, err := kcpdynamic.NewForConfig(cfg)
+	require.NoError(t, err, "error creating dynamic cluster client")
+
+	cowboysProviderKCPClient, err := kcpclientset.NewForConfig(cfg)
+	require.NoError(t, err, "error creating cowboys provider kcp client")
+
+	t.Logf("Install a cowboys APIResourceSchema into workspace %q", cowboysProviderPath)
+	mapper := restmapper.NewDeferredDiscoveryRESTMapper(memory.NewMemCacheClient(cowboysProviderKCPClient.Cluster(cowboysProviderPath).Discovery()))
+	err = helpers.CreateResourceFromFS(ctx, dynamicClusterClient.Cluster(cowboysProviderPath), mapper, nil, "apiresourceschema_cowboys.yaml", testFiles)
+	require.NoError(t, err)
+
+	// Create rolebinding to allow bind to APIExport of coboys provider. Else nobody can bind to it.
+	err = helpers.CreateResourceFromFS(ctx, dynamicClusterClient.Cluster(cowboysProviderPath), mapper, nil, "clusterrole_cowboys.yaml", testFiles)
+	require.NoError(t, err)
+
+	err = helpers.CreateResourceFromFS(ctx, dynamicClusterClient.Cluster(cowboysProviderPath), mapper, nil, "clusterrolebinding_cowboys_custom.yaml", testFiles)
+	require.NoError(t, err)
+
+	t.Logf("Create today-cowboys APIExport")
+	cowboysAPIExport := &apisv1alpha1.APIExport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "today-cowboys",
+		},
+		Spec: apisv1alpha1.APIExportSpec{
+			LatestResourceSchemas: []string{"today.cowboys.wildwest.dev"},
+			PermissionClaims: []apisv1alpha1.PermissionClaim{
+				{
+					GroupResource: apisv1alpha1.GroupResource{
+						Resource: "configmaps",
+					},
+					All: true,
+				},
+			},
+		},
+	}
+	cowboysAPIExport, err = kcpClusterClient.Cluster(cowboysProviderPath).ApisV1alpha1().APIExports().Create(ctx, cowboysAPIExport, metav1.CreateOptions{})
```

Comment:
this is async to become effective. You must put a `framework.Eventually` around the Create at the end of the test.